### PR TITLE
TT-5555 Adding OAS support

### PR DESCRIPTION
**Pull request by tbuchaillot**
### Description
This PR adds OAS support. 

If an API has `is_oas` set to `true`, Tyk-sync will use `/api/apis/oas` endpoint to update the `id`. For the API creation, the normal `/api/apis` endpoint seems enough.

We fix the OAS ID modifying the `x-tyk-api-gateway.info.dbId` field with the new ID from API creation.

### How to test it
- Use tyk-sync dump command to dump OAS and tyk APIS
Example:
`./tyk-sync dump -d="http://localhost:3000" -s="7080fcd212a6489b64f6951f4b60e7b9" -t="examples/oas"`
It should dump all the APIs correctly.

- Manually delete all the APIs from the dashboard.
- Use tyk-sync sync command to re-create the APIs
Example:
`./tyk-sync sync -d="http://localhost:3000" -s="7080fcd212a6489b64f6951f4b60e7b9" -p="examples/oas" `
It should create the API's again.

- Use tky-sync sync command again. It should update the APIs instead of deleting and creating them again.

https://tyktech.atlassian.net/browse/TT-5555

### CODE REVIEW

File: clients/dashboard/apis.go [hunk 1]
```diff
@@ -123,6 +130,9 @@ func (c *Client) CreateAPI(def *objects.DBApiDefinition) (string, error) {
 	// Create will always reset the API ID on dashboard, if we want to retain it, we must use UPDATE
 	if retainedIDs {
 		def.Id = apidef.ObjectIdHex(status.Meta)
+		if def.OAS != nil {
+
+		}
```
Comment (sredxny) [inline clients/dashboard/apis.go:135]: can we remove this?

**Review by sredxny — CHANGES_REQUESTED** [review]

File: clients/dashboard/apis.go (hunk 1 above)
Comment (tbuchaillot) [reply clients/dashboard/apis.go]: done @sredxny

**Review by sredxny — APPROVED** [review]
lgtm

File: clients/dashboard/apis.go [hunk 2]
```diff
@@ -267,9 +275,19 @@ func (c *Client) UpdateAPI(def *objects.DBApiDefinition) error {
 	asDBDef := def
 	c.fixDBDef(asDBDef)
 
-	updatePath := urljoin.Join(c.url, endpointAPIs, def.Id.Hex())
+	//data, _ := json.Marshal(asDBDef)
+	//fmt.Println(string(data))
```
Comment (matiasinsaurralde) [inline clients/dashboard/apis.go:279]: Should we remove these lines?

**Review by matiasinsaurralde — APPROVED** [review]
Looks good, added just a small comment

File: clients/dashboard/apis.go (hunk 2 above)
Comment (tbuchaillot) [reply clients/dashboard/apis.go]: whoooops, my bad 😅  removed

**Comment by tbuchaillot** [comment]
@jeffy-mathew @furkansenharputlu  can any of you guys have a look?

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**